src/TestProblem/ELBDM/PlaneWave/Init_TestProb_ELBDM_PlaneWave.cpp

```diff
@@ -0,0 +1,294 @@
+   ReadPara->Add( "PWave_XYZ",         &PWave_XYZ,             0,             0,                3                 );
+   ReadPara->Add( "PWave_LSR",         &PWave_LSR,             1,             NoMin_int,        NoMax_int         );
+
+   ReadPara->Read( FileName );
+
+   delete ReadPara;
+
+// (1-2) set the default values
+
+// (1-3) check the runtime parameters
+   if (  PWave_XYZ == 3  &&  ( amr->BoxSize[0] != amr->BoxSize[1] || amr->BoxSize[0] != amr->BoxSize[2] )  )
+      Aux_Error( ERROR_INFO, "simulation domain must be CUBIC in the %s test if PWave_XYZ == 3 !!\n", "ELBDM plane wave" );
+
+
+// (2) set the problem-specific derived parameters
+   PWave_WaveK  = 2.0*M_PI/PWave_Lambda;
+   PWave_WaveW  = 0.5*SQR( PWave_WaveK )/ELBDM_ETA;
+   PWave_Period = 2.0*M_PI/PWave_WaveW;
+   PWave_PhaseV = 0.5*PWave_WaveK/ELBDM_ETA;
+   PWave_GroupV = PWave_WaveK/ELBDM_ETA;
+
+
+
+// (3) reset other general-purpose parameters
+//     --> a helper macro PRINT_WARNING is defined in TestProb.h
+   const long   End_Step_Default = __INT_MAX__;
+   const double End_T_Default    = 6.0*PWave_Period; // 6 periods
+
+   if ( END_STEP < 0 ) {
+      END_STEP = End_Step_Default;
+      PRINT_WARNING( "END_STEP", END_STEP, FORMAT_LONG );
+   }
+
+   if ( END_T < 0.0 ) {
+      END_T = End_T_Default;
+      PRINT_WARNING( "END_T", END_T, FORMAT_REAL );
+   }
+
+
+// (4) make a note
+   if ( MPI_Rank == 0 )
+   {
+      Aux_Message( stdout, "=============================================================================\n" );
+      Aux_Message( stdout, "  test problem ID                  = %d\n",     TESTPROB_ID                );
+      Aux_Message( stdout, "  plane wave wavelength            = %13.7e\n", PWave_Lambda               );
+      Aux_Message( stdout, "  plane wave amplitude             = %13.7e\n", PWave_Amp                  );
+      Aux_Message( stdout, "  plane wave phase constant        = %13.7e\n", PWave_Phase0               );
+      Aux_Message( stdout, "  plane wave wavenumber            = %13.7e\n", PWave_WaveK                );
+      Aux_Message( stdout, "  plane wave angular frequency     = %13.7e\n", PWave_WaveW                );
+      Aux_Message( stdout, "  plane wave period                = %13.7e\n", PWave_Period               );
+      if ( PWave_LSR == 0){
+      Aux_Message( stdout, "  standing wave                    = true\n"                               );
+      }
+      else{
+      Aux_Message( stdout, "  standing wave                    = false\n"                              );
+      Aux_Message( stdout, "  plane wave propagation direction = %s%s\n",  ( PWave_LSR > 0 )  ? "+" : "-",
+                                                                           ( PWave_XYZ == 0 ) ? "x" :
+                                                                           ( PWave_XYZ == 1 ) ? "y" :
+                                                                           ( PWave_XYZ == 2 ) ? "z" : "diagonal" );
```

Comment:
Record `PWave_XYZ` for the standing wave too.

Reply:
Done.